Revert:
2016-02-04  Kyrylo Tkachov  <kyrylo.tkachov@arm.com>

	PR target/65932
	PR target/67714
	* cse.c (cse_insn): Pass NULL to fold_rtx when initially
	folding the source of a SET.
 
--- gcc/cse.c	(revision 233133)
+++ gcc/cse.c	(revision 233132)
@@ -4636,7 +4636,7 @@ cse_insn (rtx_insn *insn)
 
       /* Simplify and foldable subexpressions in SRC.  Then get the fully-
 	 simplified result, which may not necessarily be valid.  */
-      src_folded = fold_rtx (src, NULL);
+      src_folded = fold_rtx (src, insn);
 
 #if 0
       /* ??? This caused bad code to be generated for the m68k port with -O2.
